python/pyarrow/tests/test_fs.py

```diff
@@ -1107,6 +1107,10 @@ def test_s3_options():
     assert isinstance(fs, S3FileSystem)
     assert pickle.loads(pickle.dumps(fs)) == fs
 
+    fs = S3FileSystem(
+        retry_strategy=AwsStandardS3RetryStrategy(max_attempts=5))
+    assert isinstance(fs, S3FileSystem)
```

Comment:
Also test `AwsDefaultRetryStrategy`?

Reply:
Done.

Reply:
One sec, need to re-add the `ValueError` check.

Reply:
Done.